core/src/main/scala/cr/pulsar/Producer.scala

```diff
@@ -121,19 +126,39 @@ object Producer {
   }
 
   /**
-    * It creates a simple [[Producer]] with a no-op logger.
+    * It creates a [[Producer]] with a no-op logger with shared key.
     *
     * Produced messages will not be logged.
     */
-  def create[
+  def withSharedKey[
       F[_]: ContextShift: Parallel: Concurrent,
       E: Inject[*, Array[Byte]]
   ](
-      client: PulsarClient.T,
+      client: Client.T,
       topic: Topic,
       shardKey: E => MessageKey,
-      batching: Batching
+      batching: Batching = Batching.Disabled
   ): Resource[F, Producer[F, E]] =
     withLogger[F, E](client, topic, shardKey, batching, _ => _ => F.unit)
 
+  /**
+    * It creates a [[Producer]] with a no-op logger.
+    *
+    * Produced messages will not be logged.
+    */
+  def create[
+      F[_]: ContextShift: Parallel: Concurrent,
+      E: Inject[*, Array[Byte]]
+  ](
+      client: Client.T,
+      topic: Topic,
+      batching: Batching = Batching.Disabled
```

Comment:
I have mixed feelings about default values in constructors but I guess it makes it easier in this case :+1: